src/pkg/task/registry.go

```diff
@@ -0,0 +1,62 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package task
+
+import (
+	"context"
+	"fmt"
+)
+
+var (
+	checkInProcessorRegistry              = map[string]CheckInProcessor{}
+	statusChangePostFuncRegistry          = map[string]StatusChangePostFunc{}
+	executionStatusChangePostFuncRegistry = map[string]ExecutionStatusChangePostFunc{}
+)
+
+// CheckInProcessor is the processor to process the check in data which is sent by jobservice via webhook
+type CheckInProcessor func(ctx context.Context, task *Task, data string) (err error)
+
+// StatusChangePostFunc is the function called after the task status changed
+type StatusChangePostFunc func(ctx context.Context, taskID int64, status string) (err error)
+
+// ExecutionStatusChangePostFunc is the function called after the execution status changed
+type ExecutionStatusChangePostFunc func(ctx context.Context, executionID int64, status string) (err error)
+
+// RegisterCheckInProcessor registers check in processor for the specific vendor type
+func RegisterCheckInProcessor(vendorType string, processor CheckInProcessor) error {
+	if _, exist := checkInProcessorRegistry[vendorType]; exist {
+		return fmt.Errorf("check in processor for %s already exists", vendorType)
+	}
+	checkInProcessorRegistry[vendorType] = processor
```

Comment:
no data race happened here, confirm?

Reply:
No. All processors are registered when importing the packages which runs in sequence